pkg/users/accounts-list.js

```diff
@@ -18,72 +18,281 @@
  */
 
 import cockpit from 'cockpit';
-import React from 'react';
+import React, { useState } from 'react';
 import { superuser } from "superuser";
 
-import { Button, Badge, Card, CardBody, Gallery, Page, PageSection, PageSectionVariants } from '@patternfly/react-core';
-import { UserIcon } from '@patternfly/react-icons';
+import {
+    Button, Badge,
+    Card, CardHeader, CardTitle,
+    Dropdown, DropdownItem, DropdownSeparator,
+    Flex, KebabToggle, Label,
+    Page, PageSection,
+    SearchInput,
+    Text, TextVariants,
+    Toolbar, ToolbarContent, ToolbarItem
+} from '@patternfly/react-core';
+import * as timeformat from "timeformat.js";
+import { EmptyStatePanel } from 'cockpit-components-empty-state.jsx';
+import { ListingTable } from 'cockpit-components-table.jsx';
+import { SearchIcon } from '@patternfly/react-icons';
+import { SortByDirection } from "@patternfly/react-table";
 import { account_create_dialog } from "./account-create-dialog.js";
+import { delete_account_dialog } from "./delete-account-dialog.js";
+import { lockAccountDialog } from "./lock-account-dialog.js";
+import { logoutAccountDialog } from "./logout-account-dialog.js";
 
 const _ = cockpit.gettext;
 
-function AccountItem({ account, current }) {
-    function click(ev) {
-        if (!ev)
-            return;
+const UserActions = ({ account }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
 
-        if (ev.type === 'click' && ev.button !== 0)
-            return;
+    const actions = [
+        <DropdownItem key="edit-user"
+                      onClick={ev => { ev.preventDefault(); cockpit.location.go(account.name) }}>
+            {_("Edit user")}
+        </DropdownItem>,
+    ];
 
-        if (ev.type === 'keypress' && ev.key !== "Enter")
-            return;
+    superuser.allowed && actions.push(
+        <DropdownSeparator key="separator-0" />,
+        <DropdownItem key="log-user-out"
+                      isDisabled={account.uid === 0 || !account.loggedIn}
+                      onClick={() => { setKebabOpen(false); logoutAccountDialog(account) }}>
+            {_("Log user out")}
+        </DropdownItem>,
+        <DropdownSeparator key="separator-1" />,
+        <DropdownItem key="lock-account"
+                      isDisabled={account.isLocked}
+                      onClick={() => { setKebabOpen(false); lockAccountDialog(account) }}>
+            {_("Lock account")}
+        </DropdownItem>,
+        <DropdownItem key="delete-account"
+                      isDisabled={account.uid === 0}
+                      className={account.uid === 0 ? "" : "delete-account-red"}
+                      onClick={() => { setKebabOpen(false); delete_account_dialog(account) }}>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221130-151606-eb9be7bf-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)